src/pool/pool_disjoint.c

```diff
@@ -135,25 +133,32 @@ static void destroy_slab(slab_t *slab) {
         LOG_ERR("deallocation of slab data failed!");
     }
 
-    umf_ba_global_free(slab->chunks);
     umf_ba_global_free(slab);
 }
 
-// return the index of the first available chunk, SIZE_MAX otherwise
 static size_t slab_find_first_available_chunk_idx(const slab_t *slab) {
-    // use the first free chunk index as a hint for the search
-    for (bool *chunk = slab->chunks + slab->first_free_chunk_idx;
-         chunk != slab->chunks + slab->num_chunks_total; chunk++) {
-
-        // false means not used
-        if (*chunk == false) {
-            size_t idx = chunk - slab->chunks;
-            LOG_DEBUG("idx: %zu", idx);
-            return idx;
+    // Calculate the number of 64-bit words needed.
+    size_t num_words = (slab->num_chunks_total + 63) / 64;
```

Comment:
see? this calculation is exactly what `chunks_size_in_64_increments` should be. I don't get it. Why did you do it differently in two places?

Also, just store this variable in slab.